mirgecom/diffusion.py

```diff
@@ -175,44 +214,67 @@ def __init__(self, value):
         """
         self.value = value
 
-    def get_grad_flux(self, dcoll, dd_bdry, u_minus):  # noqa: D102
+    def get_grad_flux(self, dcoll, dd_bdry, kappa_minus, u_minus):  # noqa: D102
         actx = u_minus.array_context
+        kappa_tpair = TracePair(dd_bdry,
+            interior=kappa_minus,
+            exterior=kappa_minus)
         u_tpair = TracePair(dd_bdry,
             interior=u_minus,
             exterior=u_minus)
         normal = actx.thaw(dcoll.normal(dd_bdry))
-        return grad_facial_flux(u_tpair, normal)
+        return grad_facial_flux(kappa_tpair, u_tpair, normal)
 
     def get_diffusion_flux(
-            self, dcoll, dd_bdry, kappa_minus, grad_u_minus):  # noqa: D102
-        actx = grad_u_minus[0].array_context
+            self, dcoll, dd_bdry, kappa_minus, u_minus, grad_u_minus,
+            lengthscales_minus, *, penalty_amount=None):  # noqa: D102
+        actx = u_minus.array_context
         kappa_tpair = TracePair(dd_bdry,
             interior=kappa_minus,
             exterior=kappa_minus)
+        u_tpair = TracePair(dd_bdry,
+            interior=u_minus,
+            exterior=u_minus)
         normal = actx.thaw(dcoll.normal(dd_bdry))
         grad_u_tpair = TracePair(dd_bdry,
             interior=grad_u_minus,
             exterior=(
                 grad_u_minus
                 + 2 * (self.value - np.dot(grad_u_minus, normal)) * normal))
-        return diffusion_facial_flux(kappa_tpair, grad_u_tpair, normal)
+        lengthscales_tpair = TracePair(
+            dd_bdry, interior=lengthscales_minus, exterior=lengthscales_minus)
+        return diffusion_facial_flux(
+            kappa_tpair, u_tpair, grad_u_tpair, lengthscales_tpair, normal,
+            penalty_amount=penalty_amount)
+
+
+class _DiffusionKappa1Tag:
+    pass
+
+
+class _DiffusionKappa2Tag:
+    pass
 
 
-class _DiffusionStateTag:
+class _DiffusionState1Tag:
     pass
 
 
-class _DiffusionKappaTag:
+class _DiffusionState2Tag:
     pass
 
 
 class _DiffusionGradTag:
     pass
 
 
+class _DiffusionLengthscalesTag:
+    pass
+
+
 def grad_operator(
```

Comment:
@majosm , this function name struggles with this one: https://github.com/illinois-ceesd/mirgecom/blob/main/mirgecom/operators.py#L36

Maybe it would be better to rename it?

Reply:
You can alias function names when you import them, like this: https://github.com/illinois-ceesd/mirgecom/blob/24be73a00a0085996e404a74e7fb6dfc408dbcdc/mirgecom/multiphysics/thermally_coupled_fluid_wall.py#L69